client/driver/executor/executor_unix.go

```diff
@@ -53,12 +53,40 @@ func (e *UniversalExecutor) collectLogs(we io.Writer, wo io.Writer) {
 // configure new process group for child process
 func (e *UniversalExecutor) setNewProcessGroup() error {
 	// We need to check that as build flags includes windows for this file
-	if runtime.GOOS == "windows" {
-		return nil
-	}
 	if e.cmd.SysProcAttr == nil {
 		e.cmd.SysProcAttr = &syscall.SysProcAttr{}
 	}
 	e.cmd.SysProcAttr.Setpgid = true
 	return nil
 }
+
+// Cleanup any still hanging user processes
+func (e *UniversalExecutor) cleanupChildProcesses(proc *os.Process) error {
+	// If new process group was created upon command execution
+	// we can kill the whole process group now to cleanup any leftovers.
+	if e.cmd.SysProcAttr != nil && e.cmd.SysProcAttr.Setpgid {
+		if err := syscall.Kill(-proc.Pid, syscall.SIGKILL); err != nil && err.Error() != noSuchProcessErr {
+			return err
+		}
+		return nil
+	} else {
+		return proc.Kill()
+	}
+}
+
+func (e *UniversalExecutor) shutdownProcess(proc *os.Process) error {
```

Comment:
Comment that this sends the shutdown signal and doesn't necessarily kill the process